x-pack/plugin/esql/build.gradle

```diff
@@ -276,10 +273,8 @@ tasks.register("regen") {
   }
 }
 
-tasks.named("spotlessJava") { dependsOn stringTemplates }
```

Comment:
Why do we remove this dependency?

Reply:
As I was explaining [above](https://github.com/elastic/elasticsearch/pull/117899#discussion_r1867960170) `stringTemplates` always cleans the generation target directory before execution. If it is executed after other generation tasks it will remove their results now as they are in the same directory.

I had to remove it from here as spotlessJava is executed rather late. Instead I moved `stringTemplates` as early as possible.

Reply:
We cannot have multiple tasks share an output directory. If we want to have a "single" generated source directory for the purposes of the IDE then we'll need to have another task aggregate them. This should actually be a Gradle deprecation warning if you attempt to do this. I'm surprised the build didn't complain.

Reply:
its not clear to me which tasks write into src/main/generated. from what I can see in `:x-pack:plugin:esql` it is only the stringTemplate task.